### PR TITLE
Option to return all embeddings from TransformerNetwork

### PR DESCRIPTION
**Pull request by emailweixu**

### CODE REVIEW

File: alf/networks/transformer_networks.py [hunk 1]
```diff
@@ -193,4 +201,9 @@ def forward(self, inputs, state=()):
                 query = new_query
```
Comment (hnyu) [inline alf/networks/transformer_networks.py:203]: Here memory actually writes the old core, but in the docstring comment the memory writes the new core. This behavior is also different from centralized memory which always writes the new core.

File: alf/networks/transformer_networks.py [hunk 2]
```diff
@@ -161,7 +166,10 @@ def forward(self, inputs, state=()):
                 at ``__init__()``
             state (nested Tensor): states
         Returns:
-            Tensor: shape is [B, core_size * d_model]
+            - Tensor: shape is [B, core_size * d_model] if ``return_core_only``,
+                    and [B, core_size + input_size, d_model] if not ``return_core_only``,
+                    where input_size is the number of embeddings from (processed) input
```
Comment (hnyu) [inline alf/networks/transformer_networks.py]: \``input_size\``

File: alf/networks/transformer_networks.py (hunk 1 above)
Comment (emailweixu) [reply alf/networks/transformer_networks.py:203]: good catch. It's quite strange why I did this.

File: alf/networks/transformer_networks.py (hunk 2 above)
Comment (emailweixu) [reply alf/networks/transformer_networks.py]: fixed

File: alf/networks/transformer_networks.py (hunk 1 above)
Comment (hnyu) [reply alf/networks/transformer_networks.py:203]: But I think we should write the old one? because the new one will be passed as part of input next

Comment (emailweixu) [reply alf/networks/transformer_networks.py:203]: Good point. Perhaps that was what I thought originally.

Comment (hnyu) [reply alf/networks/transformer_networks.py:203]: Should also change the docstring?

Comment (emailweixu) [reply alf/networks/transformer_networks.py:203]: Already changed.

Comment (hnyu) [reply alf/networks/transformer_networks.py:203]: Also for centralized memory, we should also put old core in memory?

Comment (emailweixu) [reply alf/networks/transformer_networks.py:203]: I think putting new core into memory makes more sense for centralized memory.

**Review by hnyu — APPROVED** [review]

**Pull request merged**